packages/editor/src/plugins/jsx-blocks/Gist.js

```diff
@@ -68,8 +68,11 @@ export default ({ editor, node, attributes, props, isSelected }) => {
       {isSelected && (
         <Form
           value={props}
-          onSubmit={data => {
-            editor.setJSXProps(data)
+          onSubmit={next => {
+            editor.setJSXProps({
+              type: 'Gist',
```

Comment:
Maybe this is an okay place to keep the `data.type` intact??

Reply:
Also #13 has an example of using a generic `Form` component for different jsx block types

Reply:
Works for me, though I wonder if we should rename the function since it's doing more than setting JSX props?

Reply:
Absolutely, still getting a feel for how all this should/could fit together